x-pack/plugins/observability/public/components/threshold/threshold_rule_expression.tsx

```diff
@@ -335,11 +340,32 @@ export default function Expressions(props: Props) {
       .filter((g) => typeof g === 'string') as string[];
   }, [ruleParams, groupByFilterTestPatterns]);
 
+  if (paramsError) {
+    return (
+      <>
+        <EuiCallOut color="danger" iconType="warning" data-test-subj="thresholdRuleExpressionError">
+          <p>{paramsError.message}</p>
+        </EuiCallOut>
+        <EuiSpacer size={'m'} />
+      </>
+    );
+  }
+
+  if (!searchSource) {
+    return (
+      <>
+        <EuiEmptyPrompt title={<EuiLoadingSpinner size="xl" />} />
+        <EuiSpacer size={'m'} />
+      </>
+    );
+  }
+
   return (
     <>
       <DataViewSelectPopover
         dependencies={{ dataViews, dataViewEditor }}
         dataView={dataView}
+        metadata={{ adHocDataViewList: metadata?.adHocDataViewList || [] }}
```

Comment:
@maryam-saeidi, I can't see how we pass this adHocDataView to the rule execution, and per site it in the rule saved object 🤔

Reply:
This part was already working, I just needed to make sure the temporary data view is shown properly in the list. We handle saving it in `onSelectDataView` function.

Reply:
I checked what is the difference between the temp and the saved one. For the Temp DataView, we are persisting the map of the selected index inside the rule saved object. Where with the saved DateView we save only the reference to it.